tests/test_coordinateConverter.py

```diff
@@ -0,0 +1,556 @@
+        for cbpAltObserved in cbpAltLim:
+            self.cco.cbpAzAltObserved = SpherePoint(originalCbpAzAltObserved[0], cbpAltObserved)
+            self.assertTrue(self.cco.cbpInBounds)
+            self.assertTrue(self.cco.telInBounds)
+        for badCbpAltObserved in (
+            cbpAltLim[0] - 1e-15*radians,
+            cbpAltLim[0] - 2*degrees,
+            cbpAltLim[1] + 1e-15*radians,
+            cbpAltLim[1] + 2*degrees,
+        ):
+            self.cco.cbpAzAltObserved = SpherePoint(originalCbpAzAltObserved[0], badCbpAltObserved)
+            self.assertFalse(self.cco.cbpInBounds)
+            self.assertTrue(self.cco.telInBounds)
+        self.cco.cbpAzAltObserved = originalCbpAzAltObserved
+
+        originalTelAzAltObserved = self.cco.telAzAltObserved
+        telAltLim = self.cco.config.telAltitudeLimits
+        for telAltObserved in telAltLim:
+            self.cco.telAzAltObserved = SpherePoint(originalTelAzAltObserved[0], telAltObserved)
+            self.assertTrue(self.cco.telInBounds)
+            self.assertTrue(self.cco.cbpInBounds)
+        for badTelAltObserved in (
+            telAltLim[0] - 1e-15*radians,
+            telAltLim[0] - 2*degrees,
+            telAltLim[1] + 1e-15*radians,
+            telAltLim[1] + 2*degrees,
+        ):
+            self.cco.telAzAltObserved = SpherePoint(originalTelAzAltObserved[0], badTelAltObserved)
+            self.assertFalse(self.cco.telInBounds)
+            self.assertTrue(self.cco.cbpInBounds)
+
+    def testHolePositionsFlipX(self):
+        for telFlipX, cbpFlipX in itertools.product((False, True), (False, True)):
+            with self.subTest(telFlipX=telFlipX, cbpFlipX=cbpFlipX):
+                scc = SampleCoordinateConverter(telFlipX=telFlipX, cbpFlipX=cbpFlipX)
+                holePositions = [scc.maskInfo.getHolePos(name) for name in scc.maskInfo.holeNames]
+                if not (telFlipX or cbpFlipX):
+                    unflippedHolePositions = holePositions[:]
+                    flippedHolePositions = [(-pos[0], pos[1]) for pos in unflippedHolePositions]
+                elif cbpFlipX:
+                    self.assertPairListsAlmostEqual(holePositions, flippedHolePositions)
+                else:
+                    self.assertPairListsAlmostEqual(holePositions, unflippedHolePositions)
+
+    def testSetPupilFieldAngleFlipX(self):
+        """Test setPupilFieldAngle with varying flipX for telescope and CBP
+        """
+        beam = 3  # pick a beam that is a bit off center
+        for telFlipX, cbpFlipX in itertools.product((False, True), (False, True)):
+            with self.subTest(telFlipX=telFlipX, cbpFlipX=cbpFlipX):
+                # flip pupilPos and fieldAngle so that everything else is identical
+                # e.g. all 3-D vectors; that makes debugging easier if the test fails
+                unflippedPupilPos = (5000, -5000)
+                unflippedPupilFieldAngle = (0.1, 0.1)
```

Comment:
Is there any reason not to change this tuple and the one above to have different values for their [0] and [1] elements to also check axis inversion? _i.e._ (5000, -4000) and (0.1, 0.2) for example? Doesn't matter, but just a thought.

Reply:
Excellent suggestion